src/video_core/macro/macro.cpp

```diff
@@ -15,6 +18,20 @@
 
 namespace Tegra {
 
+namespace {
+void Dump(u64 hash, const std::vector<u32>& code) {
+    const auto base_dir{Common::FS::GetYuzuPath(Common::FS::YuzuPath::DumpDir)};
+    const auto macro_dir{base_dir / "macros"};
+    if (!Common::FS::CreateDir(macro_dir) || !Common::FS::CreateDir(base_dir)) {
+        LOG_ERROR(Common_Filesystem, "Failed to create macro dump directories");
+        return;
+    }
+    const auto name{macro_dir / fmt::format("{:016x}.macro", hash)};
+    std::fstream macro_file(name, std::ios::out | std::ios::binary);
+    macro_file.write(reinterpret_cast<const char*>(code.data()), code.size() * sizeof(u32));
```

Comment:
If you use a span for the code buffer, this can be a little more straightforward as:

```suggestion
    macro_file.write(reinterpret_cast<const char*>(code.data()), code.size_bytes());
```